Feature/41 public facing page

aggiewb opened the pull request.

aggiewb committed:
Merge pull request #38 from aggiewb/feature/28-form_to_add_new_stores

Feature/28 form to add new stores

BatemanVO committed:
Merge pull request #39 from aggiewb/release/2020.07.18

Release/2020.07.18

dependabot[bot] committed:
Bump lodash from 4.17.15 to 4.17.19

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.15 to 4.17.19.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.15...4.17.19)

Signed-off-by: dependabot[bot] <support@github.com>

BatemanVO committed:
Merge pull request #40 from aggiewb/dependabot/npm_and_yarn/lodash-4.17.19

Bump lodash from 4.17.15 to 4.17.19

aggiewb committed:
Add boilerplate html file for public facing page

aggiewb committed:
Add header section with h1 and nav element

aggiewb committed:
Add main element section with h2 elements for client

required section

aggiewb committed:
Add footer element with p element containing info about

developer creator and business copyrights

aggiewb committed:
Add title content and a tag elements in nav and footer

aggiewb committed:
Add footer to sales.html with developer creator info and

link to public facing website

aggiewb committed:
Rename css file for sales page and remove display flex

aggiewb committed:
Add client request image files

aggiewb committed:
Add resized chinook image

aggiewb committed:
Update content to include merchandise and event article

sections
Add href links to on page ids
Change main section to include h2 and h3 elements

aggiewb committed:
Add css root variables, body, img, h1, p, table, and nav style

aggiewb committed:
Remove some styling to simplify to replace with more

more responsive CSS
Remove temporarily image

aggiewb committed:
Add grid and grid-template-areas to all body contents

Remove padding-left from nav ul

aggiewb committed:
Add centering of h1 and footer and h1 font size increase

aggiewb committed:
Add merch content, update shirt mech resized img, and apply

centering

aggiewb committed:
Add font, border, and mechandise styling

aggiewb committed:
Adjust font-sizes and add border to all merch imgs

aggiewb committed:
Add h3 locations ul content with li for each location using an

address element and an added location-pin.png as bullet style
Add style for main li and address

aggiewb committed:
Correct h3 closing tag and add hours and contact content

Add body children padding and center main p

aggiewb committed:
Add mission section in main with a lorum ipsum placeholder

aggiewb committed:
Add salmon and chinook images to h1 with cooresponding

styles, and add resized salmon image

aggiewb committed:
Add form to contact us area, add form.js for interacting with

the FormSpree api, and add form styling

aggiewb committed:
Add styling for submit input, add required attribute to all

form fields, and update textarea placeholder message

aggiewb committed:
Add events ul with placeholder contents, and add events ul li

styling

aggiewb committed:
Correct href id in navbar

aggiewb committed:
Rearrange CSS for better readability and add comments

aggiewb committed:
Resize fish image and change type

aggiewb committed:
Add family image in events, and add fish image in footer

aggiewb committed:
Add css style for footer and add css animation and keyframes

for fish image in footer

aggiewb committed:
Resize family img, add styling to family img, and update h2

events content

aggiewb committed:
Add responsive styles for 700, 930, and 1200px

Adjust grid and flex styling

aggiewb committed:
Wrap products and descriptions in children and parent divs for

flex styling
Add appropriate id for accessibility

aggiewb committed:
Update font for headers, widths of header of events img, and

events img width

aggiewb committed:
Add null alt text for decorative fish images

aggiewb committed:
Add null alt tags for products as p tag describes them, and add

alt text for family image

aggiewb committed:
Correct html validator errors with textarea and alt text

aggiewb committed:
Add h2 for sales.html in footer, and remove copyright symbol

aggiewb committed:
Add margin to p tag

BatemanVO commented on css/public.css:
```diff
@@ -0,0 +1,254 @@
+}
+
+/* Contact Form */
+form > input, textarea {
```
Do you need `form > input`? Seems like every `input` in your HTML is a direct child of `form` already and we could just say `input, textarea`.

BatemanVO commented on css/public.css:
```diff
@@ -0,0 +1,254 @@
+}
+
+input[type="submit"] {
+    cursor: pointer;
```
We should combine this with the properties above, since it's the same selector.

BatemanVO commented on js/form.js:
```diff
@@ -0,0 +1,42 @@
+window.addEventListener('DOMContentLoaded', function() {
```
We should indicate that this code was provided by a third party and link to where it originated from in a comment. If available, a usage license should also be included.

BatemanVO commented on js/form.js:
```diff
@@ -0,0 +1,42 @@
+    function success() {
+        form.reset();
+        status.textContent = "Thanks! We'll be getting back to you in 24-48 hours.";
+        button.style.display = 'none';
```
We should make a CSS class with `display: none` and toggle that class on the button using `button.classList` instead of using `style.display`. Modifying the `style` property is basically the same as using `style=` in your HTML.

BatemanVO commented on js/form.js:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    function error() {
+        status.textContent = 'Oops! There was a problem.';
```
We might want to pass other properties from the `xhr` object in our `ajax` method to this function - perhaps more information about what went wrong, so that users can submit screenshots or something letting the devs know what to fix beyond "there was an issue".

BatemanVO commented on js/form.js:
```diff
@@ -0,0 +1,42 @@
+window.addEventListener('DOMContentLoaded', function() {
+    'use strict';
+
+    //Get the form elements defined in your form HTML above
```
We should remove these unnecessarily and uninformative comments that came with the original code.

BatemanVO commented on js/form.js:
```diff
@@ -0,0 +1,42 @@
+    // Handle the form submission event
+    form.addEventListener('submit', function(event) {
+        event.preventDefault();
+        const data = new FormData(form);
+        ajax(form.method, form.action, data, success, error);
```
We could combine these two lines and do `form.method, form.action, new FormData(form)` as our arguments.

I think we could also remove `success` and `error` from our arguments and parameters. The functions are already visible to the scope of the `ajax` function, so passing them in is redundant.

BatemanVO reviewed (commented):
I didn't realize there wasn't any JS on the front end page (other than the form code provided by that third-party API), that's actually pretty impressive. The markup has really good semantic tags and the responsiveness of the layouts using `grid` is great.

I think most of the improvements we could make are cleaning up code provided by someone else to begin with.

Good work!

aggiewb committed:
Combine duplicate CSS selector styles

aggiewb committed:
Remove unnecessarily and uninformative comments that came

with the original code

aggiewb committed:
Update selector as input and textarea are already direct

children of form

aggiewb committed:
Add FormSpree terms of use and link to terms of service in

commenting heading

aggiewb committed:
Use error parameter passed in to give more info to the user

about what went wrong

aggiewb committed:
Add error status code to display for users to describe issues in

more depth
Remove cached FormData() object and add it directly to ajax() call
Remove success and error from ajax arguments and parameters since they are already visible to the scope of the ajax function

aggiewb committed:
Add class for display none for form submit button

aggiewb committed:
Add button and status queries to functions they are used

Update button.style.display to use classList with displayButtonNone
Correct spelling error

aggiewb committed:
Remove trailing space
